### PR TITLE
Feat/raffles UI

### PR DESCRIPTION
**Pull request by rieg-ec**
### Contexto

Falta implementar la funcionalidad de los sorteos. En un [pr](https://github.com/platanus/lacatan/pull/59) anterior se implementó parte de la vista del show de teams, en la mayoria componentes que hacian falta.
 ​
### Qué se hizo

En este pr se implementa la ui restante para los sorteos (solo ui) y se migra a un diseño mobile y con el uso de i18n.
​
### Qué queda por hacer

Quedaria implementar la logica de los sorteos en el front y back, y arreglar el modal que en vistas mobile no se ve bien.
​
Además la parte de los links está hardcodeada porque no está muy claro qué se hará con eso. Lo dejé asi para que despues sea facil eliminarlo si no se usa, o modificarlo para que sea dinamico.

-----------
### Mobile:

Screen Shot 2021-12-14 at 13 06 55

https://user-images.githubusercontent.com/58893812/146035425-7f2fb814-530d-41f9-adb1-d488044943c7.mov

### Desktop:

Screen Shot 2021-12-14 at 13 09 32

https://user-images.githubusercontent.com/58893812/146036086-65262595-33fc-4514-b9ba-8f1650548e45.mov

### CODE REVIEW

File: engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      team_members: true
+    )
+  end
+
+  private
+
+  def tasks_definitions
+    @tasks_definitions ||= Tasks::Definition
+                           .joins(
+                             'INNER JOIN team_responsibilities ON tasks_definitions.team_responsibility_id = team_responsibilities.id
```
Comment (monkeyci) [inline engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb:15]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/LineLength: Line is too long. [133/100]

File: engines/tasks/app/serializers/tasks/api/v1/base_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Tasks::Api::V1::BaseSerializer < BaseSerializer
+  private
+
+  def nested_resource(resource, namespace = "Tasks::Api::V1")
+    super(resource, namespace)
+  end
+
+  def nested_resource(resources, namespace = "Tasks::Api::V1")
```
Comment (monkeyci) [inline engines/tasks/app/serializers/tasks/api/v1/base_serializer.rb:8]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Lint/DuplicateMethods: Method Tasks::Api::V1::BaseSerializer#nested_resource is defined at both engines/tasks/app/serializers/tasks/api/v1/base_serializer.rb:4 and engines/tasks/app/serializers/tasks/api/v1/base_serializer.rb:8.

File: engines/tasks/app/serializers/tasks/api/v1/definition_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Tasks::Api::V1::DefinitionSerializer < Tasks::Api::V1::BaseSerializer
+  type :task
+
+  attributes :id, :team_responsibility_description, :created_at, :updated_at
+
+  attribute(:team_members, if: -> { instance_options[:team_members] }) do
+    if object.team_responsibility.team.team_members
+      nested_resources(object.team_responsibility.team.team_members, namespace = nil)
```
Comment (monkeyci) [inline engines/tasks/app/serializers/tasks/api/v1/definition_serializer.rb:8]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Lint/UselessAssignment: Useless assignment to variable - namespace.

File: engines/tasks/app/serializers/tasks/base_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Tasks::BaseSerializer < BaseSerializer
+  private
+
+  def nested_resource(resource, namespace = "Tasks")
+    super(resource, namespace)
+  end
+
+  def nested_resource(resources, namespace = "Tasks")
```
Comment (monkeyci) [inline engines/tasks/app/serializers/tasks/base_serializer.rb:8]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Lint/DuplicateMethods: Method Tasks::BaseSerializer#nested_resource is defined at both engines/tasks/app/serializers/tasks/base_serializer.rb:4 and engines/tasks/app/serializers/tasks/base_serializer.rb:8.

File: app/javascript/api/team-responsibilities.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import api from './index';
+
+export default {
+  getAll(teamId) {
+    return api({
+      method: 'get',
+      url: `/api/v1/${teamId}/tasks`,
+    });
+  },
+  update(teamId, teamResponsibilityId) {
```
Comment (monkeyci) [inline app/javascript/api/team-responsibilities.js:10]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'teamId' is defined but never used                no-unused-vars

Comment (monkeyci) [inline app/javascript/api/team-responsibilities.js:10]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'teamResponsibilityId' is defined but never used  no-unused-vars

File: app/javascript/api/team-responsibilities.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    // @TODO
+  },
+  delete(teamId, teamResponsibilityId) {
```
Comment (monkeyci) [inline app/javascript/api/team-responsibilities.js:13]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'teamId' is defined but never used                no-unused-vars

Comment (monkeyci) [inline app/javascript/api/team-responsibilities.js:13]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'teamResponsibilityId' is defined but never used  no-unused-vars

File: app/javascript/components/shared/three-dot-dropdown.vue [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  mounted() {
+    const { menu, parent } = this.$refs;
+
+    this.popper = createPopper(parent, menu, {
+      placement: 'right',
+      modifiers: [
+        {
+          name: 'offset',
+          options: {
+            offset: [40, 10],
```
Comment (monkeyci) [inline app/javascript/components/shared/three-dot-dropdown.vue:55]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>No magic number: 40  no-magic-numbers

Comment (monkeyci) [inline app/javascript/components/shared/three-dot-dropdown.vue:55]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>No magic number: 10  no-magic-numbers

File: app/javascript/components/teams/team-join-leave-btn.vue [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      loading: false,
+    };
+  },
+  methods: {
+    async joinTeam() {
+      try {
+        this.loading = true;
+        await this.$store.dispatch('team/createMember', this.currentUserId);
+      } catch (e) {
+        alert(e);
```
Comment (monkeyci) [inline app/javascript/components/teams/team-join-leave-btn.vue:36]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected alert  no-alert

File: app/javascript/components/teams/team-join-leave-btn.vue [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      } finally {
+        this.loading = false;
+      }
+    },
+    async leaveTeam() {
+      try {
+        this.loading = true;
+        await this.$store.dispatch('team/deleteMember', this.currentUserId);
+      } catch (e) {
+        alert(e);
```
Comment (monkeyci) [inline app/javascript/components/teams/team-join-leave-btn.vue:46]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected alert  no-alert

File: app/javascript/components/teams/team-responsibilities.vue [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+      loading: true,
+      responsibilities: [],
+    };
+  },
+  async created() {
+    try {
+      const response = await TeamResponsibilitiesApi.getAll(this.teamId);
+      this.responsibilities = response.data.data;
+    } catch (e) {
+      alert(e);
```
Comment (monkeyci) [inline app/javascript/components/teams/team-responsibilities.vue:107]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected alert              no-alert

File: app/javascript/components/teams/team-responsibilities.vue [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    } finally {
+      this.loading = false;
+    }
+  },
+  computed: {
+    teamId() { return this.$store.state.team.team.id; },
+  },
+  methods: {
+    joinResponsibility() {
+      console.log('join resp');
```
Comment (monkeyci) [inline app/javascript/components/teams/team-responsibilities.vue:117]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected console statement  no-console

File: app/javascript/components/teams/team-responsibilities.vue [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    },
+    raffle() {
+      console.log('raffle');
```
Comment (monkeyci) [inline app/javascript/components/teams/team-responsibilities.vue:120]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected console statement  no-console

File: engines/tasks/app/serializers/tasks/api/v1/definition_serializer.rb (hunk 3 above)
Comment (rieg-ec) [reply engines/tasks/app/serializers/tasks/api/v1/definition_serializer.rb:8]: no es useless :C

File: engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+      each_serializer: Tasks::Api::V1::DefinitionSerializer,
+      team_members: true
+    )
+  end
+
+  private
+
+  def tasks_definitions
+    @tasks_definitions ||= Tasks::Definition
+      .joins('INNER JOIN team_responsibilities
```
Comment (monkeyci) [inline engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb:14]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Layout/MultilineMethodCallIndentation: Align .joins with Tasks::Definition on line 13.

File: engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+          ON tasks_definitions.team_responsibility_id = team_responsibilities.id
+        INNER JOIN teams ON team_responsibilities.team_id = teams.id
+        INNER JOIN organizations ON teams.organization_id = organizations.id')
+      .where('teams.id = ?', params[:team_id])
```
Comment (monkeyci) [inline engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb:18]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Layout/MultilineMethodCallIndentation: Align .where with Tasks::Definition on line 13.

File: engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+      .includes(team_responsibility: { team: { team_members: :user } })
```
Comment (monkeyci) [inline engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb:19]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Layout/MultilineMethodCallIndentation: Align .includes with Tasks::Definition on line 13.

File: app/javascript/components/teams/team-responsibilities.vue [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+      loading: true,
+      responsibilities: [],
+    };
+  },
+  async created() {
+    try {
+      const response = await TeamResponsibilitiesApi.getAll(this.teamId);
+      this.responsibilities = response.data.data;
+    } catch (e) {
+      alert(e);
```
Comment (monkeyci) [inline app/javascript/components/teams/team-responsibilities.vue:107]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected alert  no-alert

File: engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb (hunk 13 above)
Comment (rieg-ec) [reply engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb:14]: se queja por line too long si hago eso, se ve mas bonito asi encuentro yo (?

File: app/javascript/components/teams/team-links.vue [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+    return {
+      expanded: !(screen.width < smBreakPoint),
+      links: [
+        { platform: 'Notion', url: 'https://github.com/platanus' },
+        { platform: 'Github', url: 'https://trello.com/invite/b/HJrtFPIj/e1233f5daca0e122c7b972e6979ad1e3/golf' },
+      ],
+    };
+  },
+  mounted() {
+    window.addEventListener('resize', () => {
```
Comment (iaacosta) [inline app/javascript/components/teams/team-links.vue:51]: Acá te faltó el `event` o no?

```suggestion
    window.addEventListener('resize', (event) => {
```

Me parece que te funciona porque aún está vigente [el `Window.event`](https://developer.mozilla.org/en-US/docs/Web/API/Window/event), que es global y es el evento del `listener` actual, pero puede estar deprecado en ciertos navegadores

File: app/javascript/components/teams/team-members.vue [hunk 18]
```diff
@@ -1,101 +1,102 @@
...
+        >
+          <team-members-circles
+            :members="members"
+            :text-classes="'text-blue-500 font-semibold'"
+          />
+        </div>
+        <inline-svg
+          class="w-4 h-4"
+          @click="expand"
+          :src="require(`../../../assets/images/icons/${icon}`)"
```
Comment (iaacosta) [inline app/javascript/components/teams/team-members.vue:49]: Ídem lo de los _assets_ acá!

```suggestion
          :src="require(`assets/images/icons/${icon}`)"
```

File: app/javascript/components/teams/team-responsibilities.vue [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+    class="flex flex-col space-y-3"
+  >
+    <div class="flex flex-row justify-between w-full">
+      <span class="text-lg font-medium">{{ $t('teams.responsibilities.title') }}</span>
+      <div class="flex flex-row space-x-2">
+        <btn
+          :hierarchy="'secondary'"
+          :icon-file-name="'plus.svg'"
+          :icon-only="true"
+          :icon-classes="'w-4 h-4'"
```
Comment (iaacosta) [inline app/javascript/components/teams/team-responsibilities.vue:13]: Acá hay varios `v-bind` que te puedes ahorrar tb!

```suggestion
          hierarchy="secondary"
          icon-file-name="plus.svg"
          :icon-only="true"
          icon-classes="w-4 h-4"
```

File: app/javascript/components/teams/team-responsibilities.vue [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+                :icon-only="true"
+                :icon-file-name="'thrash.svg'"
+                :icon-classes="'w-5 h-5'"
+              />
+              <btn
+                :hierarchy="'secondary'"
+                :icon-only="true"
+                :icon-file-name="'edit.svg'"
+                :icon-classes="'w-5 h-5 fill-current'"
+              />
```
Comment (iaacosta) [inline app/javascript/components/teams/team-responsibilities.vue:72]: Ídem lo de los `v-bind`:

```suggestion
            <team-members-circles
              :members="responsibility.attributes.teamMembers"
              text-classes="font-medium"
              :text="false"
            />
            <div class="flex flex-row text-xs space-x-2 sm:text-sm">
              <btn
                @click="raffle"
                hierarchy="yellow"
                icon-file-name="ticket.svg"
              >
                <span>{{ $t('raffles.raffle') }}</span>
              </btn>
              <btn
                @click="joinResponsibility"
                hierarchy="green"
                icon-file-name="join.svg"
              >
                <span>{{ $t('raffles.join') }}</span>
              </btn>
            </div>
            <div class="flex-row hidden sm:flex space-x-2">
              <btn
                hierarchy="red"
                :icon-only="true"
                icon-file-name="thrash.svg"
                icon-classes="w-5 h-5"
              />
              <btn
                hierarchy="secondary"
                :icon-only="true"
                icon-file-name="edit.svg"
                icon-classes="w-5 h-5 fill-current"
              />
```

File: app/javascript/locales/es.js [hunk 21]
```diff
@@ -14,4 +14,39 @@ export default {
...
+    edit: {
+      message: 'Editar equipo',
+    },
+    delete: {
+      message: 'Eliminar equipo',
+      confirm: '¿Estás seguro que quieres eliminar este equipo?',
+      yes: 'Si, eliminar equipo',
+      no: 'Cancelar',
+      warn:
+        'Esta acción es irreversible y todos las tareas asociadas serán eliminadas.',
```
Comment (iaacosta) [inline app/javascript/locales/es.js:27]:
```suggestion
        'Esta acción es irreversible y todas las tareas asociadas serán eliminadas.',
```

**Review by iaacosta — CHANGES_REQUESTED** [review]
Buena! Está bacán el PR, te dejé un mix de comentarios entre chicos/medianos + de estilo/funcionales, avísame si tienes duda con cualquiera!

File: engines/tasks/app/serializers/tasks/api/v1/definition_serializer.rb (hunk 3 above)
Comment (rieg-ec) [reply engines/tasks/app/serializers/tasks/api/v1/definition_serializer.rb:8]: ahh cierto, no se me ocurrió pasar `nil` nomás

File: engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def tasks_definition
+    @tasks_definition ||= Tasks::Definition.find(params[:id])
+  end
```
[the remark this answers is not in the document]
Comment (rieg-ec) [reply engines/tasks/app/controllers/tasks/api/v1/definitions_controller.rb:24]: nop jeje

File: spec/controllers/teams_controller_spec.rb [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+  describe 'GET #show' do
+    def do_request
+      sign_in user
+      get :show, params: { id: team.id  }
+    end
+
+    before { do_request }
+
+    it { expect(response).to have_http_status(:ok) }
+  end
```
[the remark this answers is not in the document]
Comment (rieg-ec) [reply spec/controllers/teams_controller_spec.rb:28]: nunca me quedó claro si eso se testea o no en los controladores (ni tampoco qué se testea exactamente en un controlador ademas del response code jajaja), esa linea en particular seria buena idea testearla con un shared example e incluirlo en todos los tests de controlador, o hay algo mejor?

File: app/javascript/components/teams/team-links.vue [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+const smBreakPoint = parseInt(fullConfig.theme.screens.sm.replace('px', ''), 10);
+
+export default {
+  data() {
+    return {
+      expanded: !(screen.width < smBreakPoint),
+      links: [
+        { platform: 'Notion', url: 'https://github.com/platanus' },
+        { platform: 'Github', url: 'https://trello.com/invite/b/HJrtFPIj/e1233f5daca0e122c7b972e6979ad1e3/golf' },
+      ],
```
[the remark this answers is not in the document]
Comment (rieg-ec) [reply app/javascript/components/teams/team-links.vue:47]: cierto

File: app/javascript/components/teams/team-links.vue (hunk 17 above)
Comment (rieg-ec) [reply app/javascript/components/teams/team-links.vue:51]: en efecto jaja

File: app/javascript/components/teams/team-links.vue [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+    };
+  },
+  mounted() {
+    window.addEventListener('resize', () => {
+      // eslint-disable-next-line no-magic-numbers
```
[the remark this answers is not in the document]
Comment (rieg-ec) [reply app/javascript/components/teams/team-links.vue:52]: debo haber tenido uno antes y se me fue sacar la linea XD

File: app/javascript/locales/es.js (hunk 21 above)
Comment (rieg-ec) [reply app/javascript/locales/es.js:28]: buena!

File: spec/controllers/teams_controller_spec.rb (hunk 23 above)
[the remark this answers is not in the document]
Comment (rieg-ec) [reply spec/controllers/teams_controller_spec.rb]: pd: por ahora lo dejé dentro del test del controlador, pero habria que hacer lo mismo para todos los otros tests de controladores que faltan de alguna forma jajaja

File: spec/controllers/teams_controller_spec.rb [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+        do_request
+      end
+
+      it { expect(response).to have_http_status(:ok) }
+    end
+
+    context 'when user is not signed in' do
+      before { do_request }
+
+      it { expect(response).to have_http_status(302) }
```
Comment (monkeyci) [inline spec/controllers/teams_controller_spec.rb:25]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] RSpec/Rails/HttpStatus: Prefer :found over 302 to describe HTTP status code.

File: spec/controllers/teams_controller_spec.rb [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
```
Comment (monkeyci) [inline spec/controllers/teams_controller_spec.rb:46]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] RSpec/Rails/HttpStatus: Prefer :found over 302 to describe HTTP status code.

File: spec/controllers/teams_controller_spec.rb (hunk 23 above)
[the remark this answers is not in the document]
Comment (iaacosta) [reply spec/controllers/teams_controller_spec.rb]: [alerta de :mucho-texto:]

Yo generalmente a los controladores testeo que:

- Retorne los _response codes_ que espero (incluyendo `unauthorized`)
- Me retorne el/los elemento/s que espero (esto puede ser más overkill)

Para el segundo punto, te puede pasar harto que tienes una policy metida de por medio (`authorize` o `policy_scope`) y en realidad no tiene sentido testear los casos borde de eso acá, porque deberías testearlo en los tests de policy.

Entonces, para el controlador te queda hacer el stub del comportamiento de esos métodos externos e ir testeando los casos. [Acá](https://platanus.slack.com/archives/C01CEGVK0AY/p1632229387048100) hay un buen hilo de ejemplo para el `authorize`, y debería ser similar para el `policy_scope`

-----------------------

Está buena la idea del shared example! Y hay harto donde explorar y probar. Por ejemplo, el `:not_found` viene de [un _concern_ de Power Api](https://github.com/platanus/power_api/blob/5f2b3198352e3ac1e8174abd40fb159c91c5ddab/app/controllers/concerns/api/error.rb#L13-L16) luego de levantar el error de `ActiveRecord::RecordNotFound` (generalmente hecho a través de un `find` o un `find_by!`). Esto igual se repite para otros errores.

Me imagino, para el caso de autenticación, algo como esto:

```ruby
RSpec.shared_examples 'an authenticatable controller action' do
  let(:params) { nil }

  def do_request
    send(method, action, format: :json, params: params)
  end

  context 'when user is signed in' do
    let(:user) { create(:user) }

    before do
      sign_in(user)
      do_request
    end

    it "responds with #{expected_successful_status} status" do
      expect(response).to have_http_status(expected_successful_status)
    end
  end

  context 'when user is not signed in' do
    before { do_request }

    it 'responds with unauthorized status' do
      expect(response).to have_http_status(:unauthorized)
    end
  end
end
```

Y para usarlo:

```ruby
  describe '#index' do
    it_behaves_like 'an authenticatable controller action' do
      let(:method) { :get }
      let(:action) { :index }
      let(:expected_successful_status) { :ok }
    end
  end

  describe '#show' do

    it_behaves_like 'an authenticatable controller action' do
      let(:method) { :get }
      let(:action) { :index }
      let(:expected_successful_status) { :ok }
      let(params) { { id: create(:team).id } }
    end
  end
```

Quizás si uno define la función dentro del bloque del `it_behaves_like`, se pueden ahorrar esos tres `let`s (`method`, `action`, `params`)? Nunca he probado si una función pasa al _scope_ del `shared_example`, pero algo de esa línea estaría buenísimo para evitar tanto copy paste.

No creo que sea necesario ahora, pero si para un refactor general de tests si es que ninguno se hace cargo de esto.

[the remark this answers is not in the document]
Comment (rieg-ec) [reply spec/controllers/teams_controller_spec.rb]: buena! te pasaste, me quedó super claro 🥩  👏

**Review by iaacosta — APPROVED** [review]
✅

**Review by aaguilarg — APPROVED** [review]

**Pull request merged**